cmd/geth/config.go

```diff
@@ -178,6 +180,17 @@ func makeFullNode(ctx *cli.Context) (*node.Node, ethapi.Backend) {
 	}
 	backend, eth := utils.RegisterEthService(stack, &cfg.Eth)
 
+	// <specular modification>
+	log.Info("initializing specular hooks")
+	l2ChainId := ctx.Uint64(utils.NetworkIdFlag.Name)
+	l1FeeRecipient := backend.ChainConfig().L1FeeRecipient
+	if (l1FeeRecipient != common.Address{}) {
+		vm := eth.BlockChain().GetVMConfig()
+		vm.SpecularEVMPreTransferHook = hook.MakeSpecularEVMPreTransferHook(l2ChainId, l1FeeRecipient)
+		vm.SpecularL1FeeReader = hook.MakeSpecularL1FeeReader(l2ChainId)
+	}
+	// <specular modification/>
```

Comment:
@siosw stupid question, but `<specular modification> ... <specular modification />` should not rather be `<specular modification> ... </specular modification>` (assuming that's a html/xml'like block's tag convention) ?

Reply:
agree, but this is the convention used across the entire sp-geth project so far